docs/proposals/ruler-ha.md

```diff
@@ -18,51 +18,142 @@ The original issue is [#4435](https://github.com/cortexproject/cortex/issues/443
 - API
   -inconsistent API results during resharding (e.g. due to a deployment) when rulers are in a transition state loading rule groups
 
-This proposal attempts to mitigate the above risks by enabling a ruler replication factor of greater than 1, allowing multiple rulers to evaluate the same rule group — effectively, the ruler equivalent of ingester HA already supported in Cortex.
+This proposal attempts to mitigate the above risks by enabling a ruler replication factor of greater than 1, allowing multiple rulers to evaluate the same rule group — effectively, the ruler equivalent of ingester HA already supported in Cortex.  To avoid unnecessary duplication of work between replicas, we will use an elected replica strategy (again, similar to HA sample tracking) to ensure only one actual evaluation occurs for each iteration.
 
 ## Proposal
 
 ### Make ReplicationFactor configurable
 
 ReplicationFactor in Ruler is currently hardcoded to 1.  Making this a configurable parameter is the first step to enabling HA in ruler, and would also be the mechanism for the user to turn the feature on.  The parameter value will be 1 by default, equating to the feature being turned off by default.
 
-A replication factor greater than 1 will result in the same rules being evaluated by multiple ruler instances.
+A replication factor greater than 1 will result in the same rules being available for evaluation by multiple ruler instances.
 
-This redundancy will allow for missed or skipped rule evaluations from single ruler outages to be covered by other instances evaluating the same rules.
+This redundancy will allow for missed or skipped rule evaluations from single ruler outages to be covered by other instances able to evaluate the same rule groups.
 
 There is also the question of duplicate metrics generated by replicated evaluation.  We do not expect this to be a problem, as all replicas must use the same [slotted intervals](https://github.com/prometheus/prometheus/blob/b878527151e6503d24ac5b667b86e8794eb79ff7/rules/manager.go#L509) when evaluating rules, which should result in the same timestamp applying to metrics generated by each replica, with the samples being deduplicated in the ingestion path.
 
 [PR #4712](https://github.com/cortexproject/cortex/pull/4712) [open]
 
-### Adjustments for ALERTS_FOR_STATE
+### Elected ruler replicas for single iteration evaluation
 
-When an alert fires in a rulegroup shared across multiple rulers, the alert on each ruler will have slightly different timestamps due to the differences in evaluation time on each ruler.  This results in the ALERTS_FOR_STATE metric being deduplicated with the "duplicate sample for timestamp" error.  To prevent this, we synchronize Alert.activeAt for the same alerts in different rulers using a post-processing hook.
+With replication enabled, multiple rulers will evaluate the same rule group concurrently.  This will result in duplicated work, leading to duplication of the following:
```

Comment:
I think the proposal would be easier to follow if line 47 to 55 is moved to another section called "why not evaluate rules concurrently in multiple rulers"